src/screens/MyWorkouts/WorkoutSearchBar.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { useEffect, useState, useRef } from 'react';
+import { View, Text, TouchableOpacity } from 'react-native';
+import { ScrollView, TextInput } from 'react-native-gesture-handler';
+import { supabase } from '../../utils/supabaseClient';
+
+/**
+ * @param selectedExercise.exercise //value set to exercise pressed
+ * @param selectedExercise.isOk //if true, then value of exercise is valid, otherwise disregard
+ */
+export type SearchBarWorkoutProps = {
+  selectedExercise: { exercise: string | undefined; isOk: boolean | undefined };
+};
+
+const SearchBarWorkout: React.FC<SearchBarWorkoutProps> = (prop: SearchBarWorkoutProps) => {
+  const [allExercises, setAllExercises] = useState<string[]>([]);
+  const [searchInput, setSearchInput] = useState('');
+  const recentlyUsedExercise = useRef('');
+  const searchArray = useRef<string[]>([]);
+
+  if (searchInput.length > 0) {
+    searchArray.current = [];
+    const tempArr = allExercises.filter((exercise) =>
+      exercise.toLowerCase().includes(searchInput.toLowerCase())
+    );
+    tempArr.map((exercise) => {
+      searchArray.current.push(exercise);
+    });
+  }
+
+  async function fetchExercises() {
+    const { data, error } = await supabase.from('exercises').select('*');
+    if (error) {
+      alert(error);
+    }
+    if (data != null) {
+      const temp: string[] = [];
+      data.map((exercise) => {
+        temp.push(exercise.name);
+      });
+      setAllExercises(temp);
+    }
+  }
+
+  useEffect(() => {
+    fetchExercises();
+  }, []);
+
+  const onPressHandler = (item: string) => {
+    recentlyUsedExercise.current = item;
+    prop.selectedExercise.exercise = item;
+    prop.selectedExercise.isOk = true;
+    // <<----- Code to navigate back to parents page goes here ----->>
+    //
+    //
```

Comment:
This is where we want to call the callback `onSelectExercise` and pass in the exercise name. Setting the props like this won't change the parent's state.